services/policies/pkg/engine/opa/rf_mimetype.go

```diff
@@ -10,34 +12,70 @@ import (
 	"github.com/open-policy-agent/opa/types"
 )
 
-var RFMimetypeExtensions = rego.Function1(
-	&rego.Function{
-		Name:             "ocis.mimetype.extensions",
-		Decl:             types.NewFunction(types.Args(types.S), types.A),
-		Memoize:          true,
-		Nondeterministic: true,
-	},
-	func(_ rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
-		var mt string
+// RFMimetypeExtensions extends the rego dictionary with the possibility of mapping mimetypes to file extensions.
+// Be careful calling this multiple times with individual readers, the mime store is global,
+// which results in one global store which holds all known mimetype mappings at once.
+//
+// Rego: `ocis.mimetype.extensions("application/pdf")`
+// Result `[.pdf]`
+func RFMimetypeExtensions(f io.Reader) (func(*rego.Rego), error) {
+	if f != nil {
```

Comment:
Reverse logic and return an error when f is nil: 
```suggestion
	if f == nil {
	    return nil, errors.New("need reader to extend mimetypes")
```

Reply:
providing a reader is optional, if no reader is given it still works as expected